### PR TITLE
Remove tfinstall package

### LINKED ISSUE #109: Split out tfinstall into separate hcinstall module
https://github.com/hashicorp/hcinstall generalises tfinstall to all HashiCorp products with binaries on releases.hashicorp.com. It's in an experimental state with an API overhaul to improve usability.

Remember to include changes from https://github.com/hashicorp/terraform-exec/pull/111.

### PR DESCRIPTION
**Pull request by kmoe**
Closes #109.

We now import github.com/hashicorp/hc-install for installing Terraform versions during testing.

Goreleaser config is also removed, to signal that this is a library to be consumed by Go and does not compile to a binary.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -1,23 +1,4 @@
...
-    - '-s -w -X main.version={{.Version}} -X main.commit={{.Commit}}'
-    goos:
-      - linux
-      - darwin
-      - windows
-    goarch:
-      - amd64
-      - arm
-      - arm64
+project_name: terraform-exec
```
Comment (radeksimko) [inline .goreleaser.yml:1]: I'm not sure the config is useful without the `builds` section - maybe it's worth just removing it entirely?

File: go.mod [hunk 2]
```diff
@@ -4,18 +4,12 @@ go 1.14
...
 	github.com/hashicorp/go-version v1.3.0
-	github.com/hashicorp/logutils v1.0.0
+	github.com/hashicorp/hc-install v0.1.0
 	github.com/hashicorp/terraform-json v0.13.0
-	github.com/mitchellh/cli v1.1.2
 	github.com/sergi/go-diff v1.2.0
 	github.com/stretchr/testify v1.7.0
 	github.com/zclconf/go-cty v1.9.1
 	github.com/zclconf/go-cty-debug v0.0.0-20191215020915-b22d67c1ba0b
-	golang.org/x/crypto v0.0.0-20210421170649-83a5a9bb288b
```
Comment (radeksimko) [inline go.mod:20]: ❤️ 🚀

File: tfexec/internal/testutil/tfcache.go [hunk 3]
```diff
@@ -36,16 +39,32 @@ func NewTFCache(dir string) *TFCache {
...
-		return gitref.Install(ref, "", dir)
+
+	key := "gitref:" + ref
+
+	return tf.find(t, key, func(ctx context.Context) (string, error) {
+		gr := &build.GitRevision{
+			Product: product.Terraform,
+			Ref:     ref,
+		}
+		// gr.SetLogger(TestLogger())
```
Comment (radeksimko) [inline tfexec/internal/testutil/tfcache.go:50]: Was this forgotten or intentional?

File: tfexec/internal/testutil/tfcache.go [hunk 4]
```diff
@@ -36,16 +39,32 @@ func NewTFCache(dir string) *TFCache {
...
-		return f
+
+	key := "v:" + v
+
+	return tf.find(t, key, func(ctx context.Context) (string, error) {
+		ev := &releases.ExactVersion{
+			Product: product.Terraform,
+			Version: version.Must(version.NewVersion(v)),
+		}
+		// ev.SetLogger(TestLogger())
```
Comment (radeksimko) [inline tfexec/internal/testutil/tfcache.go:66]: Was this forgotten or intentional?

**Review by radeksimko — APPROVED** [review]
Just a few minor comments, non-blocking - otherwise LGTM!

File: .goreleaser.yml (hunk 1 above)
Comment (kmoe) [reply .goreleaser.yml:1]: The only thing it does now is create the GitHub release, which I agree is probably not that useful.

Comment (radeksimko) [reply .goreleaser.yml:1]: Release with what artifacts/assets? Just archive of source code?

Comment (kmoe) [reply .goreleaser.yml:1]: Yes

File: tfexec/internal/testutil/tfcache.go (hunk 3 above)
Comment (kmoe) [reply tfexec/internal/testutil/tfcache.go]: Forgotten, thanks!

File: .goreleaser.yml (hunk 1 above)
Comment (radeksimko) [reply .goreleaser.yml:1]: I think we should remove it then and avoid publishing releases if there's no "real" artifact to publish - to signal that this is a library to be consumed via Go.

**Review by radeksimko — APPROVED** [review]
:shipit:

**Pull request merged**